packages/node/src/database/NodeDiagnosticDatabaseStorageProvider.ts

```diff
@@ -0,0 +1,118 @@
+import {
+    BacktraceDatabaseConfiguration,
+    BacktraceDatabaseRecord,
+    BacktraceDatabaseStorageProvider,
+} from '@backtrace/sdk-core';
+import type { BacktraceDataBuilder } from '@backtrace/sdk-core/lib/modules/data/BacktraceDataBuilder';
+import fs from 'fs';
+import * as fsPromise from 'fs/promises';
+import path from 'path';
+import { NodeDiagnosticReportRecord } from './NodeDiagnosticReportRecord';
+
+export class NodeDiagnosticDatabaseStorageProvider implements BacktraceDatabaseStorageProvider {
+    private constructor(
+        private readonly _path: string,
+        private readonly _dataBuilder: BacktraceDataBuilder,
+        private readonly _createDatabaseDirectory: boolean = false,
+    ) {}
+
+    public static createIfValid(
+        dataBuilder: BacktraceDataBuilder,
+        options?: BacktraceDatabaseConfiguration,
+    ): BacktraceDatabaseStorageProvider | undefined {
+        if (!options) {
+            return undefined;
+        }
+
+        if (!options.enabled) {
+            return undefined;
+        }
+
+        if (options.enabled && !options.path) {
+            throw new Error(
+                'Missing mandatory path to the database. Please define the database.path option in the configuration.',
+            );
+        }
+
+        return new NodeDiagnosticDatabaseStorageProvider(options.path, dataBuilder, options.createDatabaseDirectory);
+    }
+
+    public add(): boolean {
+        return false;
+    }
+
+    public async get(): Promise<BacktraceDatabaseRecord[]> {
```

Comment:
do you have a plan when a user would set those options? Lets say the user wants to have it in the specific dir.

Reply:
I added some code to respect the paths.